examples/research_projects/codeparrot/scripts/preprocessing.py

```diff
@@ -49,18 +49,77 @@ def is_autogenerated(example, scan_width=5):
         return {"autogenerated": False}
 
 
+def is_config_test(example, scan_width=5, coeff=5):
+    """Check if file is a configuration file or a unit test by :
+    1- looking for keywords in the first few lines of the file.
+    2- counting number of occurence of the words 'config' and 'test' with respect to number of lines.
+    """
+
+    keywords = ["unit tests", "test file", "configuration file"]
+    lines = example["content"].splitlines()
+    count_config = 0
+    count_test = 0
+    # first test
+    for _, line in zip(range(scan_width), lines):
+        for keyword in keywords:
+            if keyword in line.lower():
+                return {"config_test": True}
+    # second test
+    nlines = example["content"].count("\n")
+    threshold = coeff if nlines < 300 else coeff * (nlines // 300)
```

Comment:
Why is the behaviour different before and after 300? It feels a bit weird to have that hardcoded. Could we have a global threshold?

Reply:
I use 300 as a limit to determine short files that use coeff as threshold, maybe I'll pass it as an argument to the function

Reply:
Is it necessary to make that distinction between short and long files? E.g. if the coeff is 0.1 (1 occurrence in every 10 lines) do we need to make a different rule if the file has less than 100 lines? 

This would be much easier to explain and document:
```python
threshold = coeff * n_lines
```

What do you think?

Reply:
Ah I see what you mean, we can do `threshold =constant+ int(coeff *nlines)` And `constant` will be applied if ever the second term is 0 (for example for setup files that are usually short)🙂